#!/usr/bin/env python
PACKAGE = 'leg_detector'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add('connection_threshold',     double_t,   0, '[m]', 0.10, 0, .25)
gen.add('min_points_per_group',     int_t,      0, '', 5, 1, 20)


gen.add('leg_reliability_limit',    double_t,   0, '', 0.8, 0, 1)
gen.add('people_probability_limit', double_t,   0, '', 0.8, 0, 1)

gen.add('publish_leg_measurements',             bool_t,     0, '', True)
gen.add('publish_leg_velocity',             bool_t,     0, '', True)
gen.add('publish_people',           bool_t,     0, '', True)
gen.add('publish_leg_markers',      bool_t,     0, '', True)
gen.add('publish_people_markers',   bool_t,     0, '', False)
gen.add('publish_clusters',         bool_t,     0, '', True)
gen.add('publish_particles',        bool_t,     0, '', True)
gen.add('publish_matches',          bool_t,     0, '', True)
gen.add('publish_leg_history',      bool_t,     0, '', True)
gen.add('publish_people_tracker',   bool_t,     0, '', True)
gen.add('publish_leg_feature_arrows',   bool_t,     0, '', True)
gen.add('publish_static_people_trackers',   bool_t,     0, '', True)

gen.add('publish_occlusion_model',   bool_t,     0, '', True)

gen.add('no_observation_timeout',   double_t,   0, 'Timeout tolerance for no observations [s]', 0.5, 0, 5)
gen.add('max_second_leg_age',       double_t,   0, '[s]', 2.0, 0, 5)
gen.add('max_track_jump',           double_t,   0, '[m]', 1.0, 0, 5)
gen.add('max_meas_jump',            double_t,   0, '[m]', 0.75, 0, 5)
gen.add('leg_pair_separation',      double_t,   0, '[m]', 1.0, 0, 2)
gen.add('fixed_frame',              str_t,      0, 'Fixed Frame', 'odom_combined')

exit(gen.generate(PACKAGE, 'dual_tracker', 'DualTracker'))
